lib/msf/base/sessions/meterpreter.rb

```diff
@@ -255,9 +255,10 @@ def reset_ui
   def kill
     begin
       cleanup_meterpreter
-      self.sock.close
+      self.sock.close if self.sock
```

Comment:
If there's a possibility of re-entrancy here, should we `self.sock = nil` after?